### PR TITLE
trust-dns-rustls: enable early data on tokio-rustls crate

### PR DESCRIPTION
**Pull request by daareiza**
Enable early data on the `tokio-rustls` dependency, otherwise client config will be completely ignored leading to undesired results.

### CODE REVIEW

**Comment by bluejekyll** [comment]
This feature isn't well documented in the tokio-rustls crate. I think it would be worth adding some documentation to that crate about this.

From the docs:

```
    /// Enable 0-RTT.
    ///
    /// Note that you want to use 0-RTT.
    /// You must set `enable_early_data` to `true` in `ClientConfig`.
```

I think this is a good change, though it would be nice if we had some tests that verified `0-RTT` is functioning. I haven't really investigated what we need to do in order to make `0-RTT` work.

@quininer if you have a moment could you comment?

**Comment by quininer** [comment]
I think this is a good fit for `trust-dns`, because dns is basically idempotent so using 0-RTT does not affect security.

I don't know what dot or doh server supports 0-RTT. maybe cloudflare support it, but I have not tried it.

**Comment by bluejekyll** [comment]
We could make sure it's enabled in the trust-dns-server implementations, and add a test for it somehow? We don't have to do it as part of this change, but would you recommend a strategy for that?

**Comment by quininer** [comment]
Unfortunately, `rustls` does not support the early data of server side. using `openssl` backend might be a good idea.

**Comment by quininer** [comment]
For the client, we only need to set [ClientConfig::enable_early_data](https://docs.rs/rustls/0.16.0/rustls/struct.ClientConfig.html#structfield.enable_early_data) and [TlsConnector::early_data](https://github.com/quininer/tokio-rustls/blob/master/src/lib.rs#L99) to true, and tokio-rustls will handle fallback automatically.

File: Cargo.lock [hunk 1]
```diff
@@ -1293,10 +1293,11 @@ dependencies = [
 
 [[package]]
 name = "tokio-rustls"
-version = "0.12.0-alpha.5"
+version = "0.12.0-alpha.7"
```
Comment (quininer) [inline Cargo.lock]: I just released 0.12.0-alpha.8.

**Comment by bluejekyll** [comment]
Thanks for the PR!

**Pull request merged**